### PR TITLE
sauce-connect: 4.3.16 -> 4.4.0

### PR DESCRIPTION
**Pull request by lo1tuma**
##### Things done
- Built on platform(s)
  - [x] OS X
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

[Changelog](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+Version+4.4.0)

### CODE REVIEW

**Comment by gilligan** [comment]
I just tested `sc` on NixOS ✅

**Comment by LnL7** [comment]
the patch phase should probably be, instead of reordering the phases.

```
patchPhase = stdenv.lib.optionalString stdenv.isLinux ''
    patchelf \
      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
      --set-rpath "$out/lib:${makeLibraryPath [zlib]}" \
      $out/bin/sc
  '';
```

**Comment by lo1tuma** [comment]
@gilligan Thank you for testing this on NixOS :simple_smile:.

@LnL7 Good suggestion, I’ve applied the changes to this PR 😉 .

File: pkgs/development/tools/sauce-connect/default.nix [hunk 1]
```diff
@@ -4,27 +4,24 @@ with lib;
...
   buildInputs = [ unzip ];
-  phases = [ "unpackPhase" ]
-   ++ (lib.optionals (stdenv.system != "x86_64-darwin") [ "patchPhase" ])
-   ++ [ "installPhase " ];
 
-  patchPhase = ''
+  patchPhase = stdenv.lib.optionalString stdenv.isLinux ''
     patchelf \
       --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
       --set-rpath "$out/lib:${makeLibraryPath [zlib]}" \
```
Comment (LnL7) [inline pkgs/development/tools/sauce-connect/default.nix]: I think `$out/lib` doesn't exist

Comment (gilligan) [inline pkgs/development/tools/sauce-connect/default.nix]: @LnL7 so as discussed on irc the `./lib` folder does actually exist right? :)

**Pull request merged**